### PR TITLE
feat(styleguide): Allow component level links

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

* Let's you specify `component="Box"` et al on our custom storybook internal link which sends you to our best guess story.  This may not work for components that are not directly specified as a `component={X}` on a stories' meta configuration.

Usage:
```tsx
<LinkTo component="Box">Box</LinkTo>
```

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@43.2.2-alpha.e9afdf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61168c39020d0400d66535fb--gamut-preview.netlify.app>https://61168c39020d0400d66535fb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61168c39020d0400d66535fb>Deploy Logs</a></p>

File: packages/styleguide/.storybook/components/Markdown/Elements.tsx [hunk 1]
```diff
@@ -33,15 +34,29 @@ type LinkProps = Omit<ComponentProps<typeof Anchor>, 'variant'> & {
...
+  ...props
+}) => {
+  let storyId = id;
+  const componentLinks = useComponentLinks();
+
+  if (component) {
+    storyId = componentLinks[component] ?? id;
   }
+
+  const computedProps = {} as Pick<LinkProps, 'onClick'>;
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/components/Markdown/Elements.tsx:51]: Unused?

```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great, thanks!

I bet it's going to be a long time till we have two components with the same name... maybe that should be considered an error case?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@43.2.2-alpha.e6e7bd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6116a4600b917d198d532042--gamut-preview.netlify.app>https://6116a4600b917d198d532042--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6116a4600b917d198d532042>Deploy Logs</a></p>

**Pull request merged**